app/Models/Foundation/Summit/Registration/SummitRegistrationInvitation.php

```diff
@@ -701,4 +704,53 @@ public function setAcceptanceCriteria(string $acceptance_criteria):void{
             throw new ValidationException(sprintf("acceptance_criteria %s is not allowed.", $acceptance_criteria));
         $this->acceptance_criteria = $acceptance_criteria;
     }
+
+    /**
+     * @return string
+     */
+    public function getStatus(): string
+    {
+        return $this->status;
+    }
+
+    /**
+     * @param string $status
+     * @throws ValidationException
+     */
+    public function setStatus(string $status): void
+    {
+        if(!in_array($status,self::AllowedStatus))
+            throw new ValidationException(sprintf("status %s is not allowed.", $status));
+
+        if ($this->status === self::Status_Pending)
+            $this->action_date = new \DateTime('now', new \DateTimeZone('UTC'));
+
+        $this->status = $status;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isAccepted(): bool
+    {
+        return $this->status === self::Status_Accepted;
+    }
+
+    /**
+     * @param bool $accepted
+     * @throws \Exception
+     */
+    public function setAccepted(bool $accepted)
```

Comment:
dont use booleans just tell
create a markAccepted and markRejected method ( flag over object code smell)